diff --git b/plugins/vpp/binapi/ip/ip.ba.go a/plugins/vpp/binapi/ip/ip.ba.go
index 6564081d..a68ca172 100644
--- b/plugins/vpp/binapi/ip/ip.ba.go
+++ a/plugins/vpp/binapi/ip/ip.ba.go
@@ -340,8 +340,8 @@ type FibPath struct {
 	NextHopID         uint32
 	RpfID             uint32
 	ViaLabel          uint32
-	NLabels           uint8
-	LabelStack        []FibMplsLabel `struc:"[16]FibMplsLabel"`
+	NLabels           uint8 `struc:"sizeof=LabelStack"` // MANUALLY FIXED, see https://jira.fd.io/browse/VPP-1261
+	LabelStack        []FibMplsLabel
 }

 func (*FibPath) GetTypeName() string {
